internal/corazawaf/rulegroup.go

```diff
@@ -128,7 +130,15 @@ RulesLoop:
 		// Rules with phase 0 will always run
 		if r.Phase_ != 0 && r.Phase_ != phase {
 			// Execute the rule in inferred phases too if multiphase evaluation is enabled
-			if !multiphaseEvaluation || !r.inferredPhases.has(phase) {
+			// For chained rules, inferredPhases is not relevant, we rather have to run from minimal potentially
+			// matchable phase up to the rule's defined phase (chainMinPhase <= phase <= Phase_)
+			// At the first run chainMinPhase is not set, so we look at the parent chain rule's minimal phase.
+			// If it is not reached, we skip the whole chain, there is no chance to match it.
+			if !multiphaseEvaluation ||
+				(!r.HasChain && !r.inferredPhases.has(phase)) ||
+				(r.HasChain && phase < r.chainMinPhase) ||
+				(r.HasChain && !r.inferredPhases.hasOrMinor(phase) && !r.withPhaseUnknownVariable) ||
```

Comment:
Here I added yet another control to reduce chained rules evaluated at impossible times. hasOrMinor is an extended way to look at the inferredPhases of the parent rule, returning true not only if the specific phase is inferred, but also if any other previous phase was already an inferred phase.
For example:
```
SecRule REQUEST_URI "/chain_phase2" "id:10, phase:2, t:none, log, setvar:'tx.set1=1', chain"
	SecRule REQUEST_BODY "chain_phase2" "setvar:'tx.set2=2', chain"
		SecRule REQUEST_BODY "chain_phase2" "setvar:'tx.set3=3'" 
```
Even if the parent rule has a single variable at phase:1, during phase 2 these rule has to be considered ready to be evaluated again participating in the match of the whole chain. 
Vice versa, if the parent rule has variables that are not yet ready, there is no point in evaluating the chain because it will never match (this is the control enforced here).